client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,366 @@
+
+		menu_item = Gtk.MenuItem.new_with_label('Delete')
+		menu_item.connect('activate', self.signal_menu_activate_delete_prompt)
+		self.popup_menu.append(menu_item)
+		self.popup_menu.show_all()
+
+	def _signal_treeview_button_pressed(self, _, event):
+		if event.button == Gdk.BUTTON_SECONDARY:
+			self.popup_menu.popup(None, None, None, None, event.button, Gtk.get_current_event_time())
+			return True
+		return
+
+	def signal_expand_row(self, _, treeiter, treepath):
+		new_path = self._tvmodel[treeiter][2]  # pylint: disable=unsubscriptable-object
+		self.load_dirs(new_path, treeiter)
+		self._tvmodel.remove(self._tvmodel.iter_children(treeiter))
+
+	def signal_collapse_row(self, _, treeiter, treepath):
+		current = self._tvmodel.iter_children(treeiter)
+		while current:
+			self._tvmodel.remove(current)
+			current = self._tvmodel.iter_children(treeiter)
+		self._tvmodel.append(treeiter, [None, None, None, None, None, None, None])
+
+	def load_dirs(self, path, parent=None):
+		counter = 0
+		for name in self._yield_dir_list(path):
+			if self.local_hidden and name.startswith('.'):
+				continue
+			fullname = path + '/' + name
+			perm = self._check_perm(fullname)
+			raw_time = self._get_raw_time(fullname)
+			date_modified = '   ' + time.strftime('%y-%m-%d %H:%M:%S', time.localtime(raw_time))
+			is_folder = self._get_is_folder(fullname)
+			if is_folder:
+				icon = Gtk.IconTheme.get_default().load_icon('folder', 20, 0)
+				if perm[3] != 'r' and perm[5] != 'x':
+					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, None, -1, date_modified))
+			else:
+				file_size = self._get_file_size(fullname)
+				hr_file_size = '   ' + boltons.strutils.bytes2human(file_size)
+				if perm[3] != 'r' and perm[5] != 'x':
+					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
+				icon = Gtk.IconTheme.get_default().load_icon('empty', 12.5, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, hr_file_size, file_size, date_modified))
+			if is_folder and perm[3] == 'r' and perm[5] == 'x':
+				self._tvmodel.append(current, [None, None, None, None, None, None, None])
+			counter += 1
+		if counter < 1:
+			self._tvmodel.append(parent, [None, None, None, None, None, None, None])
+
+	def signal_menu_toggled_hidden_files(self, _):  # pylint: disable=method-hidden
+		self.local_hidden = not self.local_hidden
+		self.refresh()
+
+	def refresh(self):
+		model = self._tvmodel
+		exp_lines = []
+		model.foreach(lambda model, path, iter: exp_lines.append(path) if self.treeview_local.row_expanded(path) else 0)
+		self.treeview_local.collapse_all()
+		for path in exp_lines:
+			self.treeview_local.expand_row(path, False)
+
+	def signal_menu_activate_delete_prompt(self, _):
+		selection = self.treeview_local.get_selection()
+		model, treeiter = selection.get_selected()
+		dialog = Gtk.Dialog('Warning')
```

Comment:
This should use the king phisher [show_dialog_yes_no](https://github.com/securestate/king-phisher/blob/master/king_phisher/client/gui_utilities.py#L452) function.